File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //ask the user how much change is owed
    float c;
    do
    {
        c = get_float("How much change do you need?: ");
    }
    while (c < 0);

    //round up the change
    c = round(c * 100);

    //count up to change
    int n = 0;

    while (0 < c)
    {
        if (25 <= c)
        {
            c -= 25;
            n++;
        }
        else if (10 <= c)
        {
            c -= 10;
            n++;
        }
        else if (5 <= c)
        {
            c -= 5;
            n++;
        }
        else if (1 <= c)
        {
            c -= 1;
            n++;
        }
    }

    printf("%i\n", n);
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    c = prompt()

    c = round(c * 100)

    n = 0
    while (0 < c):
        if (25 <= c):
            c -= 25
            n += 1
        elif (10 <= c):
            c -= 10
            n += 1
        elif (5 <= c):
            c -= 5
            n += 1
        elif (1 <= c):
            c -= 1
            n += 1

    print(n)


def prompt():
    c = 0
    while True:
        c = get_float("How mush do I owe you?")
        if (c > 0):
            break
    return c


main()